eclipse-sirius/sirius-components
# [1253] Fix the computation of unsynchronized semantic elements in studios

Bug: https://github.com/eclipse-sirius/sirius-components/issues/1253
Signed-off-by: Pierre-Charles David <pierre-charles.david@obeo.fr>

# Pull request template

## General purpose
What is the main goal of this pull request?

## Project management

## Architectural decision records (ADR)

## Dependencies

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- Hooks are properly typed:

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose

### Architecture

## Review

### How to test this PR?

* Import the following studio: [issue-1253-studio.zip](https://github.com/eclipse-sirius/sirius-components/files/8862343/issue-1253-studio.zip)
* In a separate project, create an instance model with Container -> E1 -> E2
* Create a diagram on the container. E1 (and only E1) should be visible.
* Drag E2 from the explorer inside the E1 node on the diagram.
* A node representing E2 should appear inside E1 on the diagram.
